kind-demo-super-heroes/README.md

```diff
@@ -0,0 +1,178 @@
+
+> db.Fights.update({"_id":  ObjectId("6621237ca09c6f6f5c647a24")}, {$set: {"winnerName": "luca"}})
+WriteResult({ "nMatched" : 1, "nUpserted" : 0, "nModified" : 1 })
+> 
+```
+
+13. Explain that we are going to update the 'rest fights' microservice to increase the chances for the weaker fighter to win. Currently, both fighters receive extra random bonus points, which almost always results in the weaker fighter losing. We want to change this so that a weaker fighter could be motivated to beat a stronger opponent and win.
+
+14. Open the project in VSCode and show the app code.
+15. Copy and paste the code to update the behavior of the `determineWinner` function within the `quarkus-super-heroes/rest-fights/src/main/java/io/quarkus/sample/superheroes/fight/service/FightsService.java` file. We will modify the code so that the weaker fighter receives an additional 1 to [opponent's level] points added to their level.
+
+```
+Uni<Fight> determineWinner(@SpanAttribute("arg.fighters") FightRequest fightRequest) {
+Log.debugf("Determining winner between fighters: %s", fightRequest);
+
+        // Amazingly fancy logic to determine the winner...
+        return Uni.createFrom().item(() -> {
+                Fight fight;
+
+                int heroLevel = fightRequest.hero().level();
+                int villainLevel = fightRequest.villain().level();
+
+                if (heroLevel > villainLevel) {
+                        int extraMotivation = this.random.nextInt(heroLevel);
+                        villainLevel += extraMotivation;
+                } else if (heroLevel < villainLevel) {
+                        int extraMotivation = this.random.nextInt(villainLevel);
+                        heroLevel += extraMotivation;
+                }
+
+                if (heroLevel > villainLevel) {
+                        fight = heroWonFight(fightRequest);
+                        fight.winnerLevel = heroLevel;
+                        fight.loserLevel = villainLevel;
+                } else if (heroLevel < villainLevel) {
+                        fight = villainWonFight(fightRequest);
+                        fight.winnerLevel = villainLevel;
+                        fight.loserLevel = heroLevel;
+                } else {
+                        fight = getRandomWinner(fightRequest);
+                }
+
+                fight.fightDate = Instant.now();
+                
+                return fight;
+        });
+}
+```
+
+16. We compile the quarkus app by using `mvnw clean package -Dnative -DskipTests=true -Dquarkus.native.container-build=true` where `-Dquarkus.native.container-build=true` is needed to create a native linux executable.
+
+N.B: on Windows we must first initialize the Microsoft Native Tools for Visual Studio -> https://quarkus.io/guides/building-native-image.html.
```

Comment:
Is that required as we are building in container ?

Reply:
If i don't use the native tools console i face 
```
Unknown lifecycle phase ".native.container-build=true". You must specify a valid lifecycle phase or a goal in the format <plugin-prefix>:<goal> or <plugin-group-id>:<plugin-artifact-id>[:<plugin-version>]:<goal>. Available lifecycle phases are: pre-clean, clean, post-clean, validate, initialize, generate-sources, process-sources, generate-resources, process-resources, compile, process-classes, generate-test-sources, process-test-sources, generate-test-resources, process-test-resources, test-compile, process-test-classes, test, prepare-package, package, pre-integration-test, integration-test, post-integration-test, verify, install, deploy, pre-site, site, post-site, site-deploy. ->
```